web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -257,8 +315,20 @@ const Resources = memo(
           props.updateUserPreferences({ unifiedResourcePreferences })
         }
         pinning={pinning}
-        resources={resources.map(mapToSharedResource)}
-        resourcesFetchAttempt={attempt}
+        resources={resources.map(r => {
+          const { resource, ui } = mapToSharedResource(r);
+          return {
+            resource,
+            ui: {
+              ActionButton: props.getAccessRequestButton(r) || ui.ActionButton,
```

Comment:
I'd have expected this logic to be implemented on the side of the shared component. 🤔 But idk, maybe it's fine here.

Reply:
But then we would have to pass both `ActionButton` and `AccessRequestButton` to the shared component and we have places (new request page in the Web UI ent) where we only provide request buttons. Because of that, I think it's better to give the callistes more flexibility.